chain/beacon/beacon.go

```diff
@@ -42,10 +44,10 @@ type BeaconPoint struct {
 type RandomBeacon interface {
 	Entry(context.Context, uint64) <-chan Response
 	VerifyEntry(types.BeaconEntry, types.BeaconEntry) error
-	MaxBeaconRoundForEpoch(abi.ChainEpoch) uint64
+	MaxBeaconRoundForEpoch(network.Version, abi.ChainEpoch) uint64
```

Comment:
Shouldn't we use https://github.com/filecoin-project/lotus/pull/7389/files#diff-587eaf0df6b60dbf741d19bbe39439f6f48ffe171e82a71c76b82484ba48f386R23-R38 for determining the right version ? Now this PR is putting the both logic (1) select the right version (2) return the right randomness in the same method impacting all functions, while the Schedule seems to have been made to clearly separate the logics so it's easy to upgrade later on (without changing necessarily the API of  RandomBeacon or the rest of the code)

Reply:
Link doesn't work :/

Reply:
but reflecting nikkolasg comment, should we just use beacon schedule and upgrade the beacon to another "version" of itself, instead of adding this in?